crypto/bn/asm/mips.pl

```diff
@@ -488,8 +488,9 @@
 	$ST	$ta3,-2*$BNSZ($a0)
 
 	.set	noreorder
-	bgtz	$ta0,.L_bn_sqr_words_loop
 	$ST	$ta2,-$BNSZ($a0)
+	bgtz	$ta0,.L_bn_sqr_words_loop
+	 nop
```

Comment:
Does `$ST` really expand into multiple instructions?  Or is gcc issuing a misleading warning?
(Ditto for `$LD` below...)

Reply:
I am not a mips expert but see http://math-atlas.sourceforge.net/devel/assembly/mips-iv.pdf section A 2.1.2.

Reply:
Well, I'm not an assembly programmer on any CPU, so don't really know what the right answer is, but if the GCC warning is legit, then a comment could be useful here.  Or else a comment about why it is OK to ignore the GCC warning.  Perhaps
@levitte will have more constructive feedback here.  The rest of the PR looks fine (module the cut/paste that I expect will be fixed soon).

Reply:
@dot-asm I hope you might find a few minutes to comment the GCC warnings about the MIPS branch delay slot. Is there in fact a potential issue with:
```
    bgtz	$ta0,.L_bn_sqr_words_loop
    $ST	$ta2,-$BNSZ($a0)
```
that is mitigated by instead writing:
```
    $ST	$ta2,-$BNSZ($a0)
    bgtz	$ta0,.L_bn_sqr_words_loop
    nop
```
on some MIPS systems?

Reply:
Just in case to eliminate ambiguity, $ST is always **one** mnemonic, `sw` or `sd` depending on processor bit-ness, so it's not about script, but assembler itself. Secondly, even though MIPS assemblers are known to support synthetic mnemonics, such that get expanded into multiple instructions, loads and stores are **not** such mnemonics, they map to actual instructions 1-to-1. Now, I don't quite catch one thing. It sounds like it's about warning. But if I try to trigger it [with GNU assembler up to 2.34], I fail with all ABI variants. So is there any specific assembler flag that was used? In other words how was it compiled more specifically?

On side note one has to recognize that if it was an actual error, then it wouldn't stand 10 years of not failing (even though it wasn't tested as frequently since I retired my SGI workstation).

Reply:
I was unable to trigger it either for both mips and mips 64. I will revert.